plugin/trace/trace.go

```diff
@@ -81,34 +89,29 @@ func (t *trace) Name() string { return "trace" }
 
 // ServeDNS implements the plugin.Handle interface.
 func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	trace := false
-	if t.every > 0 {
-		queryNr := atomic.AddUint64(&t.count, 1)
-
-		if queryNr%t.every == 0 {
-			trace = true
-		}
-	}
 	span := ot.SpanFromContext(ctx)
-	if !trace || span != nil {
+	if span != nil {
 		return plugin.NextOrFailure(t.Name(), t.Next, ctx, w, r)
 	}
 
 	req := request.Request{W: w, Req: r}
-	span = t.Tracer().StartSpan(spanName(ctx, req))
+	span = t.Tracer().StartSpan("servedns")
 	defer span.Finish()
 
 	rw := dnstest.NewRecorder(w)
 	ctx = ot.ContextWithSpan(ctx, span)
 	status, err := plugin.NextOrFailure(t.Name(), t.Next, ctx, rw, r)
 
+	span.SetTag(tagClient, req.IP())
 	span.SetTag(tagName, req.Name())
-	span.SetTag(tagType, req.Type())
+	span.SetTag(tagProto, req.Proto())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
+	span.SetTag(tagType, req.Type())
+	if t.EndpointType == "datadog" {
```

Comment:
Datadog may expect some extra attributes to really look good (Example: span.type). @gbbr may know.

Reply:
I didn't find anything worthwhile in [ddtrace-ext](https://godoc.org/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext)
But i might have missed an obvious one.

Reply:
Correct, `span.type` will be helpful.

Reply:
From the previously mentioned godoc 
```
(web, db, cache)
```
Are the one that are available

Can i assume that `web` is for something that has some sort of CRUD interface ?
Or `cache` would be more in line with the DNS inner working ?

Reply:
We could map the DNS response to HTTP code.
But imho that will look really clunky.

Reply:
There is `SpanTypeDNS` which I think would be appropriate.